lib/puppet/provider/registry_value/registry.rb

```diff
@@ -174,6 +174,57 @@ def write_value
     end
   end
 
+
+  def data_to_bytes(type, data, &blk)
+    bytes = []
+
+    case type
+      when Win32::Registry::REG_SZ, Win32::Registry::REG_EXPAND_SZ
+        bytes = wide_string(data).bytes
+      when Win32::Registry::REG_MULTI_SZ
+        # each wide string is already NULL terminated
+        bytes = data.map { |s| wide_string(s).bytes }.flat_map { |a| a }
+        # requires an additional NULL terminator to terminate properly
+        bytes << 0 << 0
+      when Win32::Registry::REG_BINARY
+        # TODO: not sure what to do here - old code did data.to_s
+        bytes = wide_string(data).bytes
```

Comment:
Yeah, I hadn't yet debugged this part to make sure I understand what was getting passed in based on the `to_native` call - but I was pretty sure the above wasn't right.

Looks like `data.bytes` is probably the right thing - the `to_s` part I mention in the comment is from the Ruby impl at: 
https://github.com/ruby/ruby/blob/v2_1_6/ext/win32/lib/win32/registry.rb#L736-L737